confluence-slack-server-integration-plugin/src/main/java/com/atlassian/confluence/plugins/slack/spacetochannel/service/DefaultSpacesWithAdminPermissionProvider.java

```diff
@@ -139,15 +133,12 @@ public DefaultSpacesWithAdminPermissionProvider(
         return builder.build();
     }
 
-    private String queryOrderBy(Session session) {
-        final HibernateUtil.SQLDialect dialect = HibernateUtil.getDialect(session);
-        LOGGER.debug("Using {} as dialect for space query");
-
-        switch (dialect) {
-            case MSSQL:
-                return QUERY_ORDER_LENGTH_MSSQL;
-            default:
-                return QUERY_ORDER_LENGTH;
+    private String queryOrderBy() {
+        String dialect = systemInformationService.getDatabaseInfo().getDialect();
+        if (dialect != null && dialect.matches(".*?MySQL.*?Dialect$")) {
```

Comment:
1. Is usage of regex actually needed here? I suppose it may be replaced with easier .contains("MsSql").
**No its not needed. I will simplify this.**
2. In original version this condition worked for MS SQL database detection, while now - for MySql. They are not same. Could you also add the full name of the expected matched dialect name in the comments? Or a link to the documentation with list of available dialects. If it changes in future, we'll have to change this logic to support more patterns.
You are right. I will correct the DB Name.  I will try to compile a list of dialect names too.